openwisp_controller/connection/models.py

```diff
@@ -25,8 +25,17 @@ class ConnectorMixin(object):
     _connector_field = 'connector'
 
     def clean(self):
+        # Validate the connector field here to avoid ImportError in case that it is not valid
+        # (eg. it is an empty string because the user has forgotten to pick up one from the choices)
+        # The field value is validated in a later stage anyways. We are returning {} to avoid showing
+        # a duplicate error message for the field
+        if not self._get_connector():
+            raise ValidationError({})
         self._validate_connector_schema()
 
+    def _get_connector(self):
+        return getattr(self, self._connector_field)
```

Comment:
please update https://github.com/openwisp/openwisp-controller/blob/36c3b7d6d90621808fa50fe52e5aa58222aed8de/openwisp_controller/connection/models.py#L53 to use this new method so we avoid duplication